File: C++/DynLib/plString.cpp
#include <string.h>
#include "plString.h"

plString::plString() {
    str = "";
}

plString::plString(plString * init) {
    int len = strlen(init->str);
    str = new char[len+1];
    strcpy(str, init->str);
}

plString::plString(const char* init) {
    int len = strlen(init);
    str = new char[len+1];
    strcpy(str, init);
}

plString::~plString() { delete str; }

const char* plString::value() { return str; }
void plString::set(const char* value) {
    delete str;
    int len = strlen(value);
    str = new char[len+1];
    strcpy(str, value);
}

int plString::length() { return strlen(str); }

void plString::read(hsStream * S) {
    delete str;
    str = S->readSafeStr();
}

void plString::write(hsStream * S) {
    S->writeSafeStr(str);
}

const char* plString::getTypeName() { return ""; }
const int plString::getTypeNum() { return -1; }

File: Java/Plasma/DynLib/plString.java
package Plasma.DynLib;

import Plasma.plCreatable;
import Plasma.CoreLib.hsStream;
import java.io.IOException;

public class plString extends plCreatable {
    public String str;

    public plString() { str = ""; }
    public plString(plString init) { str = new String(init.str); }
    public plString(String init) { str = new String(init); }

    public void read(hsStream S) throws IOException {
        str = S.readSafeString();
    }
    public void write(hsStream S) throws IOException {
        S.writeSafeString(str);
    }

    public int getTypeNum() { return -1; }
    public String getTypeName() { return ""; }
}

